chain/network/src/peer_manager.rs

```diff
@@ -739,43 +767,29 @@ impl Handler<NetworkRequests> for PeerManagerActor {
                 NetworkResponses::NoResponse
             }
             NetworkRequests::BlockRequest { hash, peer_id } => {
-                if let Some(active_peer) = self.active_peers.get(&peer_id) {
-                    active_peer
-                        .addr
-                        .do_send(SendMessage { message: PeerMessage::BlockRequest(hash) });
+                if self.send_message(ctx, &peer_id, PeerMessage::BlockRequest(hash)) {
+                    NetworkResponses::NoResponse
+                } else {
+                    NetworkResponses::RouteNotFound
```

Comment:
Could all `send_` methods return `NetworkResponses::NoResponse` or `NetworkResponses::RouteNotFound`? It may help to avoid a series of "if self.send... { } else { }".
The comments may be like "returns `NetworkResponses::NoResponse` if message is successfully sent or `NetworkResponses::RouteNotFound` otherwise".

Reply:
I see those `if` are somehow cumbersome, but doesn't feel well to return a `NetworkResponse`. To avoid that many if I can return an enum { `Sent`, `Dropped` } and implement into `NetworkResponse`